jupyter-on-gke/tests/test_hub.py

```diff
@@ -0,0 +1,64 @@
+import sys
+import requests
+import yaml
+
+from packaging.version import Version as V
+
+
+def test_hub_up(hub_url):
+    r = requests.get(hub_url)
+    r.raise_for_status()
+    print("Jupyterhub up.")
+
+
+def test_api_root(hub_url):
+    """
+    Tests the hub api's root endpoint (/). The hub's version should be returned.
+
+    A typical jupyterhub logging response to this test:
+
+        [I 2019-09-25 12:03:12.051 JupyterHub log:174] 200 GET /hub/api (test@127.0.0.1) 9.57ms
+    """
+    r = requests.get(hub_url + "/hub/api")
+    r.raise_for_status()
+    info = r.json()
+    assert V("4") <= V(info["version"]) <= V("5")
+    print("Jupyterhub Rest API is working.")
+
+
+def test_hub_login(hub_url):
```

Comment:
@yiyinglovecoding As we discussed offline, can we create a separate tracking bug to test the actual IAP auth flow, once you separate the ray and jupyter tests? FYI the current blocker is that provisioning the managed cert for IAP takes 20-30 mins, but since this is a presubmit test, it should be ok to wait for the cert to provision.

Reply:
will do in separate PR